### PR TITLE
[POT-7] Display company by ID

### PR DESCRIPTION
**Pull request by pkondratas**
Created an API method through which a company with a specified ID is retrieved. Unit and integration tests are also present. Upon successful request, company details are returned. Otherwise status code of a request is 404.

### CODE REVIEW

File: template/src/main/java/pot/insurance/manager/controller/CompanyRestController.java [hunk 1]
```diff
@@ -23,4 +24,9 @@ public CompanyDTO createCompany(@RequestBody CompanyDTO companyDTO) {
     public Object getAllCompanies() {
```
Comment (anton-kolyaev) [inline template/src/main/java/pot/insurance/manager/controller/CompanyRestController.java:24]: Change returning type to `List<CompanyDTO>` please - I've missed that in previous review

File: template/src/main/java/pot/insurance/manager/controller/UserRestController.java [hunk 2]
```diff
@@ -31,7 +31,6 @@ public UserDTO saveUser(@RequestBody UserDTO userDTO) {
     @GetMapping()
     public Object findAllUsers(){
```
Comment (anton-kolyaev) [inline template/src/main/java/pot/insurance/manager/controller/UserRestController.java:32]: And here to` List<UserDTO>` and further to `UserDTO`

File: template/src/main/java/pot/insurance/manager/service/CompanyService.java [hunk 3]
```diff
@@ -44,4 +44,14 @@ public Object getAllCompanies() {
             .map(companyMapper::companyToCompanyDTO)
             .toList();
     }
+
+    public CompanyDTO getCompanyById(UUID companyId) {
+        Optional<Company> company = companyRepository.findById(companyId);
+
+        if(company.isPresent()) {
```
Comment (anton-kolyaev) [inline template/src/main/java/pot/insurance/manager/service/CompanyService.java:51]: Use `Optional.orElseThrow()` instead
https://www.geeksforgeeks.org/optional-orelsethrow-method-in-java-with-examples/

**Review by anton-kolyaev — CHANGES_REQUESTED** [review]

**Comment by pkondratas** [comment]
Made the requested changes, resolved merge conflicts.

**Review by anton-kolyaev — APPROVED** [review]

**Review by NorthDragonDuck — APPROVED** [review]

**Pull request merged**